src/components/menu/menu.svelte

```diff
@@ -77,19 +83,28 @@
     // a change event.
     if (!item) return
 
+    // Use data-is-interactive=true to prevent the menu from closing when selected. This infers
+    // there is interacitivity inside the menu item (e.g. a Toggle), which would be good for the user
+    // to see change state and allowing the user to manually close when ready.
+    if (
+      (item.tagName === 'LEO-OPTION' || item.tagName === 'LEO-MENU-ITEM') &&
+      !item.dataset.isInteractive
```

Comment:
I think it would be ideal if we didn't have to mark items as `data-interactive=true` - it complicates the API somewhat - is it not possible to use some combination of `stopPropagation` and `preventDefault` to achieve the same result?

Reply:
We still want to have `item.click()` fire. In other words, we want to be able to hit kbd arrows and then enter to fire the click event. Perhaps since we're piggybacking on element focus then "enter" might cause the element's onclick to fire? I can't remember. Overall this is probably less complicated if we make Components out of child items (since they seem to have particular design patterns) and expose attributes that affect this explicitly. Or some other way to expose options more cleanly instead of a mix of tag name, data-attributes (and class names) - data attributes everywhere?! 🙃